rcl_lifecycle/include/rcl_lifecycle/rcl_lifecycle.h

```diff
@@ -430,6 +430,25 @@ RCL_LIFECYCLE_PUBLIC
 void
 rcl_print_state_machine(const rcl_lifecycle_state_machine_t * state_machine);
 
+/// Print the transition map
+/**
+ * This function will print in the standard output the transition map
```

Comment:
It's actually logging it not just printing it to stdout. I think the logging could go other places too. 

Given that, maybe formatting it into a string and letting the user print it is one option, or actually printing it to stdout, or update the docs and leave it logging to info, understanding that the user won't be able to control if it goes to info instead of debug, for example.